### PR TITLE
dev-python/black: new package

### PR DESCRIPTION
**Pull request by TheChymera**
Package provides standardized Python formatting

Package-Manager: Portage-2.3.99, Repoman-2.3.22
Signed-off-by: Horea Christian <chr@chymera.eu>

### CODE REVIEW

File: dev-python/black/black-19.10_beta_p0.ebuild [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+DESCRIPTION="The uncompromising Python code formatter"
+HOMEPAGE="https://github.com/psf/black"
+SRC_URI="mirror://pypi/${PN:0:1}/${PN}/${PN}-${MY_PV}.tar.gz"
+
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="test"
+
+DEPEND="
```
Comment (thesamesam) [inline dev-python/black/black-19.10_beta_p0.ebuild:22]: Are these actually BDEPENDs?

File: dev-python/black/black-19.10_beta_p0.ebuild [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+	test? (
+		dev-python/aiohttp[${PYTHON_USEDEP}]
+		dev-python/aiohttp-cors[${PYTHON_USEDEP}]
+		)
```
Comment (thesamesam) [inline dev-python/black/black-19.10_beta_p0.ebuild:26]: Indentation

File: dev-python/black/black-19.10_beta_p0.ebuild [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+	dev-python/attrs[${PYTHON_USEDEP}]
+	dev-python/appdirs[${PYTHON_USEDEP}]
+	dev-python/toml[${PYTHON_USEDEP}]
+	dev-python/typed-ast[${PYTHON_USEDEP}]
+	dev-python/regex[${PYTHON_USEDEP}]
+	dev-python/pathspec[${PYTHON_USEDEP}]
+	$(python_gen_cond_dep 'dev-python/dataclasses[${PYTHON_USEDEP}]' 'python3_6' )
+	dev-python/typing-extensions[${PYTHON_USEDEP}]
+	dev-python/mypy_extensions[${PYTHON_USEDEP}]
+	"
```
Comment (thesamesam) [inline dev-python/black/black-19.10_beta_p0.ebuild:39]: Indentation

File: dev-python/black/black-19.10_beta_p0.ebuild [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+MY_PV="${MY_PV/_p/}"
+
+DESCRIPTION="The uncompromising Python code formatter"
+HOMEPAGE="https://github.com/psf/black"
+SRC_URI="mirror://pypi/${PN:0:1}/${PN}/${PN}-${MY_PV}.tar.gz"
+
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="test"
```
Comment (thesamesam) [inline dev-python/black/black-19.10_beta_p0.ebuild:20]: Do you need a RESTRICT for this?

File: dev-python/black/black-19.10_beta_p0.ebuild (hunk 1 above)
Comment (TheChymera) [reply dev-python/black/black-19.10_beta_p0.ebuild:22]: no

File: dev-python/black/black-19.10_beta_p0.ebuild (hunk 4 above)
Comment (TheChymera) [reply dev-python/black/black-19.10_beta_p0.ebuild:20]: In what case would I need that?

Comment (thesamesam) [reply dev-python/black/black-19.10_beta_p0.ebuild:20]: You only want to run tests if USE=test, right?

File: dev-python/black/black-19.10_beta_p0.ebuild (hunk 1 above)
Comment (thesamesam) [reply dev-python/black/black-19.10_beta_p0.ebuild:22]: Maybe I'm misunderstanding?

[16:30:53]  <sam_c> chymera: You've lost me, I'm afraid
[16:31:01]  <sam_c> chymera: Surely all test dependencies are only BDEPENDs?
[16:32:02]  <sam_c> https://devmanual.gentoo.org/general-concepts/dependencies/index.html
[16:32:09]  <sam_c> "BDEPEND specifies dependencies applicable to CBUILD, i.e. programs that need to be executed during the build, "

File: dev-python/black/black-19.10_beta_p0.ebuild (hunk 4 above)
Comment (TheChymera) [reply dev-python/black/black-19.10_beta_p0.ebuild:20]: no, tests are triggered by `FEATURES=test`

Comment (Whissi) [reply dev-python/black/black-19.10_beta_p0.ebuild:20]: @thesamesam was right. You need
```bash
RESTRICT="!test? ( test )"
```
Please see https://devmanual.gentoo.org/general-concepts/dependencies/#test-dependencies for more information.

**Comment by TheChymera** [comment]
@Alessandro-Barbieri the tests pass just fine over here, and in any case, if the Gentoo `dev-python/aiohttp-cors` package is broken, that's an issue best addressed with that package, not with its deps.

**Comment by TheChymera** [comment]
ok, apparently this got added by someone else already.

**Pull request closed without merging**